### PR TITLE
[StableHLO] Use stablehlo submodule

### LINKED ISSUE #12678: Sever IREE repo connection with mlir-hlo
Issues with IREE's mlir-hlo dependency have been blocking integrates for 2 weeks. We've shared the intention to cut ties with this dependency, and now we need to do so expediently to unblock the integrates.

Very high-level description of what's needed:
- Create proposal, making sure to confirm with @GMNGeoffrey and @jpienaar that the proposal is coherent when integrating IREE back into Google.
- Extract some files out of mlir-hlo, organize them appropriately in IREE, delete submodule. 
- Submit when tests pass. 

More detailed plan should be discussed with Magnus Hyttsten (not sure his github handle).

### PR DESCRIPTION
**Pull request by kuhar**
Use the stablehlo submodule to provide stablehlo, since we do not need the 'mhlo' part or mlir-hlo anymore.

Fixes: https://github.com/openxla/iree/issues/12678

### CODE REVIEW

File: build_tools/third_party/stablehlo/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Copyright 2020 The IREE Authors
```
Comment (kuhar) [inline build_tools/third_party/stablehlo/CMakeLists.txt]: 2023

**Comment by kuhar** [comment]
The failing lint check is for some existing issues in `build_tools/scripts/integrate/patch_module.py`. This PR only modified a comment so it shouldn't require formatting changes there.

**Comment by kuhar** [comment]
Ping

File: CMakeLists.txt [hunk 2]
```diff
@@ -807,7 +807,7 @@ else()
 
   # Add default external projects.
   iree_llvm_add_external_project(mlir-iree-dialects ${CMAKE_CURRENT_SOURCE_DIR}/llvm-external-projects/iree-dialects)
-  iree_llvm_add_external_project(mlir-hlo ${CMAKE_CURRENT_SOURCE_DIR}/third_party/mlir-hlo)
+  iree_llvm_add_external_project(stablehlo ${CMAKE_CURRENT_SOURCE_DIR}/third_party/stablehlo)
   if(IREE_INPUT_TORCH)
     iree_llvm_add_external_project(torch-mlir-dialects ${CMAKE_CURRENT_SOURCE_DIR}/third_party/torch-mlir-dialects)
   endif()
```
Comment (ScottTodd) [inline CMakeLists.txt:813]: Keeping this switch simple for now is fine, but we might want to put `stablehlo` behind `IREE_INPUT_STABLEHLO` like how `torch-mlir-dialects` is behind `IREE_INPUT_TORCH`

File: build_tools/scripts/integrate/iree_modules.py [hunk 3]
```diff
@@ -30,14 +30,14 @@ def __init__(self, *, name: str, path: str, branch_pin_file: str,
...
-        default_repository_url="https://github.com/iree-org/iree-mhlo-fork.git",
-        fork_repository_push="git@github.com:iree-org/iree-mhlo-fork.git",
-        fork_repository_pull="https://github.com/iree-org/iree-mhlo-fork.git",
-        branch_prefix="patched-mlir-hlo-",
+        name="stablehlo",
+        path="third_party/stablehlo",
+        branch_pin_file="third_party/stablehlo.branch-pin",
+        default_repository_url="https://github.com/iree-org/iree-stablehlo-fork.git",
+        fork_repository_push="git@github.com:iree-org/iree-stablehlo-fork.git",
+        fork_repository_pull="https://github.com/iree-org/iree-stablehlo-fork.git",
```
Comment (ScottTodd) [inline build_tools/scripts/integrate/iree_modules.py]: Just `stablehlo`, not `iree-stablehlo-fork`?
```suggestion
        default_repository_url="https://github.com/iree-org/stablehlo.git",
        fork_repository_push="git@github.com:iree-org/stablehlo.git",
        fork_repository_pull="https://github.com/iree-org/stablehlo.git",
```

**Comment by ScottTodd** [comment]
CODEOWNERS added lots of reviewers for this. I pruned the list a bit, but it might also help to do that sooner / explicitly say who you expect reviews from.

**Review by antiagainst — APPROVED** [review]
LGTM; just need to address Scott's comments there.

**Comment by burmako** [comment]
Wow this is awesome! Very happy to know that this works seamlessly after https://github.com/openxla/stablehlo/pull/1612.

File: CMakeLists.txt (hunk 2 above)
Comment (kuhar) [reply CMakeLists.txt:813]: Yeah, I want to avoid changing too many things at one

File: build_tools/scripts/integrate/iree_modules.py (hunk 3 above)
Comment (kuhar) [reply build_tools/scripts/integrate/iree_modules.py]: Ah, my sed failed me here. Thanks for catching this, @ScottTodd!

**Review by ScottTodd — APPROVED** [review]

**Comment by antiagainst** [comment]
Awesome, congrats!!

**Pull request merged**